webui/react/src/components/kit/Select.tsx

```diff
@@ -0,0 +1,163 @@
+import { Select as AntdSelect } from 'antd';
+import type { DefaultOptionType, LabeledValue, RefSelectProps, SelectValue } from 'antd/es/select';
+import React, { forwardRef, useCallback, useMemo, useState } from 'react';
+
+import Label, { LabelTypes } from 'components/Label';
+import Icon from 'shared/components/Icon/Icon';
+
+import css from './Select.module.scss';
+
+const { OptGroup, Option } = AntdSelect;
+
+export { Option, SelectValue };
+
+type Options = DefaultOptionType | DefaultOptionType[];
+export interface Props<T = SelectValue> {
+  allowClear?: boolean;
```

Comment:
`SelectProps` would be a good name to follow other UIkit naming convention
Also, generics should be `T extends SelectValue` instead of `T = SelectValue`

Reply:
Agreed, good feedback I have made the changes.

Reply:
looks like need to be `T extends SelectValue = SelectValue`

Reply:
question:  do we make `Select` component generics or not for now?

Reply:
I am unsure what you mean by making the "`Select` components generic" could you give me some sort of example or more detailed explanation?

Reply:
What i meant  is that `SelectProps` generic is always `SelectValue` cuz `Select` component doesn't accept type from outside like `<Select<someType>>{children}</Select>`.  In [the previous comment](https://github.com/determined-ai/determined/pull/6102#discussion_r1122454008), i said generics could be removed because of the reason that the component doesn't accept generics. 
Doesnt have to be in this PR, but i think `Select` component should be like something like that (pseudo-code )

```tsx
const Select = <T>({{...}: SelectProps<T>) {...}
```